File: src/main/java/ink/ptms/raphael/Raphael.java
package ink.ptms.raphael;

import io.izzel.taboolib.loader.Plugin;
import io.izzel.taboolib.module.config.TConfig;
import io.izzel.taboolib.module.inject.TInject;
import net.milkbowl.vault.permission.Permission;
import org.bukkit.Bukkit;
import org.bukkit.plugin.ServicePriority;

@Plugin.Version(5.15)
public final class Raphael extends Plugin {

    @TInject
    public static TConfig CONF = null;

    @Override
    public void onLoading() {
        Bukkit.getServicesManager().register(Permission.class, new RaphaelHook(), getPlugin(), ServicePriority.Normal);
    }

    @Override
    public void onStopping() {
        Bukkit.getServicesManager().unregister(RaphaelAPI.INSTANCE.getPermission());
    }
}

File: src/main/java/ink/ptms/raphael/Raphael.kt
package ink.ptms.raphael

import ink.ptms.raphael.RaphaelAPI.permission
import net.milkbowl.vault.permission.Permission
import org.bukkit.Bukkit
import org.bukkit.plugin.ServicePriority
import taboolib.common.platform.Plugin
import taboolib.module.configuration.Config
import taboolib.module.configuration.SecuredFile
import taboolib.platform.BukkitPlugin

object Raphael : Plugin() {

    @Config
    lateinit var conf: SecuredFile
        private set

    override fun onEnable() {
        Bukkit.getServicesManager().register(Permission::class.java, RaphaelHook(), BukkitPlugin.getInstance(), ServicePriority.Normal)
    }

    override fun onDisable() {
        Bukkit.getServicesManager().unregister(permission)
    }
}